app/models/agents/local_file_agent.rb

```diff
@@ -0,0 +1,186 @@
+      MD
+    end
+
+    event_description do
+      "Events will looks like this:\n\n    %s" % if boolify(interpolated['watch'])
+        Utils.pretty_print(
+          "file_pointer" => {
+            "file" => "/tmp/test/filename",
+            "agent_id" => id
+          },
+          "event_type" => "modified/added/removed"
+        )
+      else
+        Utils.pretty_print(
+          "file_pointer" => {
+            "file" => "/tmp/test/filename",
+            "agent_id" => id
+          }
+        )
+      end
+    end
+
+    def default_options
+      {
+        'mode' => 'read',
+        'watch' => 'true',
+        'append' => 'false',
+        'path' => "",
+        'data_path' => '$.data'
+      }
+    end
+
+    form_configurable :mode, type: :array, values: %w(read write)
+    form_configurable :watch, type: :array, values: %w(true false)
+    form_configurable :path, type: :string
+    form_configurable :append, type: :boolean
+    form_configurable :data_path, type: :string
+
+    def validate_options
+      if options['mode'].blank? || !['read', 'write'].include?(options['mode'])
+        errors.add(:base, "The 'mode' option is required and must be set to 'read' or 'write'")
+      end
+      if options['watch'].blank? || ![true, false].include?(boolify(options['watch']))
+        errors.add(:base, "The 'watch' option is required and must be set to 'true' or 'false'")
+      end
+      if options['append'].blank? || ![true, false].include?(boolify(options['append']))
+        errors.add(:base, "The 'append' option is required and must be set to 'true' or 'false'")
+      end
+      if options['path'].blank?
+        errors.add(:base, "The 'path' option is required.")
+      end
+    end
+
+    def working?
+      should_run?(false) && ((interpolated['mode'] == 'read' && check_path_existance && checked_without_error?) ||
+                             (interpolated['mode'] == 'write' && received_event_without_error?))
+    end
+
+    def check
+      return if interpolated['mode'] != 'read' || boolify(interpolated['watch']) || !should_run?
+      return unless check_path_existance(true)
+      if File.directory?(interpolated['path'])
+        Dir.glob(File.join(interpolated['path'], '*')).select { |f| File.file?(f) }
+      else
+        [interpolated['path']]
+      end.each do |file|
+        create_event payload: get_file_pointer(file)
+      end
+    end
+
+    def receive(incoming_events)
+      return if interpolated['mode'] != 'write' || !should_run?
+      incoming_events.each do |event|
+        mo = interpolated(event)
+        File.open(mo['path'], boolify(mo['append']) ? 'a' : 'w') do |file|
+          file.write(Utils.value_at(event.payload, mo['data_path']))
+        end
+      end
+    end
+
+    def start_worker?
+      interpolated['mode'] == 'read' && boolify(interpolated['watch']) && should_run? && check_path_existance
```

Comment:
When the options change the agents worker will have a different [worker_id](https://github.com/cantino/huginn/blob/master/app/concerns/long_runnable.rb#L41), then the old worker will be [killed](https://github.com/cantino/huginn/blob/master/lib/agent_runner.rb#L73) and restarted.